src/xunit.assert/Asserts/Sdk/AssertEqualityComparer.cs

```diff
@@ -149,6 +158,36 @@ public bool Equals(T x, T y)
             return dictionaryYKeys.Count == 0;
         }
 
+        bool? CheckIfSetsAreEqual(T x, T y, TypeInfo typeInfo)
+        {
+            if (!IsSet(typeInfo))
+                return null;
+
+            var enumX = x as IEnumerable;
+            var enumY = y as IEnumerable;
+            if (enumX == null || enumY == null)
+                return null;
+
+            var objectEnumX = enumX.Cast<object>();
+            var objectEnumY = enumY.Cast<object>();
+            if (objectEnumX.Count() != objectEnumY.Count())
+                return false;
+
+            var setX = new HashSet<object>(objectEnumX, objectComparer);
```

Comment:
So I'm probably missing something, but I can't find a way to get the underlying element type as GetGenericArguments is not available in xunit.assert and xunit.core, and also I don't see a way to use it after creating an instance of AssertEqualityComparer of that type since I can't get a reference to the Comparer (of the correct type). 
```cs
var comparerType = typeof(AssertEqualityComparer<>).MakeGenericType(typeof(T).GetGenericArguments()/*unavailable*/);

object/*Of what type?*/ comparer = Activator.CreateInstance(comparerType);
```
Any ideas?